storage/innobase/xtrabackup/src/redo_log.cc

```diff
@@ -65,9 +65,9 @@ bool Redo_Log_Reader::find_start_checkpoint_lsn() {
       mach_read_from_8(log_sys->checkpoint_buf + LOG_CHECKPOINT_NO);
 
   while (true) {
-    err = fil_io(IORequest(IORequest::READ), true,
-                 page_id_t(log_sys->files_space_id, 0), univ_page_size, 0,
-                 LOG_FILE_HDR_SIZE, log_hdr_buf, nullptr);
+    err = fil_redo_io(IORequest(IORequest::LOG | IORequest::READ),
```

Comment:
b184bd30f94df30a8bf178fc327590c5865d33bc added an assertion to enforce IO on redo is done via fil_redo_io:
https://github.com/altmannmarcelo/percona-xtrabackup/blob/PXB-2690-8.0/storage/innobase/fil/fil0fil.cc#L8475
```
. . .
dberr_t fil_io(const IORequest &type, bool sync, const page_id_t &page_id,
               const page_size_t &page_size, ulint byte_offset, ulint len,
               void *buf, void *message) {
  auto shard = fil_system->shard_by_id(page_id.space());
#ifdef UNIV_DEBUG
  if (!sync) {
    /* In case of async io we transfer the io responsibility to the thread which
    will perform the io completion routine. */
    static_cast<buf_page_t *>(message)->release_io_responsibility();
  }
#endif
  /* For redo IO the fil_redo_io must be used. */
  ut_ad(shard->id() != REDO_SHARD);
  ```